engine/src/main/java/org/terasology/engine/subsystem/rpc/DiscordRPCSubSystem.java

```diff
@@ -0,0 +1,308 @@
+        if (lastRichPresence == null) {
+            RichPresence.Builder builder = new RichPresence.Builder();
+            builder.setLargeImage(LARGE_IMAGE);
+            lastRichPresence = builder.build();
+        }
+        client.sendRichPresence(lastRichPresence);
+    }
+
+    @Override
+    public void onDisconnect(IPCClient client, Throwable t) {
+        if (ready) {
+            ready = false;
+        }
+        getLogger().info("Discord RPC >> Disconnected!");
+    }
+
+    @Override
+    public void run() {
+        while (autoReconnect) {
+            try {
+                // Ignore if the Discord RPC is not enabled
+                if (!enabled) {
+                    if (ready) {
+                        try {
+                            getInstance().ipcClient.close();
+                        } catch (Exception ex) { }
+                    }
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Don't try to retry to discover the RPC until some reasons happen
+                if (dontTryAgain) {
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Connect if the connect on init didn't connect successfully
+                if (!connectedBefore && !ready) {
+                    lastPing = 0;
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) { }
+                    Thread.sleep(15 * 1000);
+                    if (!ready) {
+                        reconnectTries += 1;
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                    }
+                    continue;
+                }
+
+
+                // Ping to make sure that the RPC is alive
+                if (ready) {
+                    Thread.sleep(1);
+                    lastPing += 1;
+                    if (lastPing >= 5 * 1000) {
+                        ipcClient.sendRichPresence(this.lastRichPresence);
+                        this.lastPing = 0;
+                    }
+                } else { // Reconnect to RPC with tries ( min: 2 seconds, max: 30 seconds )
+                    lastPing = 0;
+                    reconnecting = true;
+                    int timeout = (reconnectTries * 2) * 1000;
+                    getLogger().info("Discord RPC >> Reconnecting... (Timeout: " + timeout + "ms)");
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) {
+                        if (reconnectTries <= 5) {
+                            reconnectTries += 1;
+                        }
+                        if (reconnectTries >= 5) {
+                            dontTryAgain = true;
+                        }
+                        Thread.sleep(timeout);
+                    }
+                }
+            } catch (Exception ex) { }
```

Comment:
Might want a comment explaining why the exception is ignored 👍

Reply:
**Done** I added a clearly stuff to handle the errors and ignore the `InterruptedException`.